src/sardana/macroserver/recorders/storage.py

```diff
@@ -137,11 +137,11 @@ def _startRecordList(self, recordlist):
             all_motors = self.macro.findObjs('.*', type_class=Type.Motor)
             all_motors.sort()
             for mot in all_motors:
-                pos = mot.getPosition()
+                pos = mot.getPosition(force=True)
                 if pos is None:
                     record = "%s = nan\n" % (mot)
                 else:
-                    record = "%s = %g\n" % (mot, mot.getPosition())
+                    record = "%s = %g\n" % (mot, mot.getPosition(force=True))
```

Comment:
Is it necessary to read it again? Since it is not None, it should have the value already.
I would use: `record = "%s = %g\n" % (mot, pos)`